$NetBSD: patch-af,v 1.1 2001/02/22 11:34:59 agc Exp $

> From Wofgang Solfrank (ws@tools.de)
> What I discovered was a problem in ckddasd.c, where there are two lseeks
> with an offset of -(CKDDASD_RECHDR_SIZE).  Now CKDDASD_RECHDR_SIZE is
> defined (in hercules.h) as sizeof(CKDDASD_RECHDR), which happens to be
> an unsigned int.  Since the offset in the lseek (and in the ckd_lseek
> that is used as an interface routine to it) is of type off_t, i.e. 64-bit,
> the unsigned is zero-extended to 64 bits, and thus the sign is lost.

--- ckddasd.c	2001/02/22 10:54:55	1.1
+++ ckddasd.c	2001/02/22 10:55:26
@@ -1663,7 +1663,7 @@
     }
 
     /* Backspace over end of track marker */
-    rc = ckd_lseek (dev, dev->fd, -(CKDDASD_RECHDR_SIZE), SEEK_CUR);
+    rc = ckd_lseek (dev, dev->fd, (int) -(CKDDASD_RECHDR_SIZE), SEEK_CUR);
     if (rc == -1)
     {
         /* Handle seek error condition */
@@ -1813,7 +1813,7 @@
     }
 
     /* Backspace over end of track marker */
-    rc = ckd_lseek (dev, dev->fd, -(CKDDASD_RECHDR_SIZE), SEEK_CUR);
+    rc = ckd_lseek (dev, dev->fd, (int) -(CKDDASD_RECHDR_SIZE), SEEK_CUR);
     if (rc == -1)
     {
         /* Handle seek error condition */
